Dashboard enable and validate dashboard login Tests

sunilangadi2 opened the pull request:
Signed-off-by: Sunil Angadi <sangadi@redhat.com>

# Description
Jira ticket: https://projects.engineering.redhat.com/browse/RHCEPHQE-183

In bootstrap with **skip-dashboard** deploys ceph with dashboard module disabled.
So added test case to enable the dashboard and validate dashboard login with provided credentials

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs

# Checklist:

- [x] Create a test case in Polarion reviewed and approved.
- [x] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [x] Review the automation design
- [x] Implement the test script and perform test runs
- [x] Submit PR for code review and approve
- [x] Update Polarin Test with Automation script details and update automation fields
- [x] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test

sunilkumarn417 reviewed (commented):
Thanks for this PR and few minor comments to make generic improvements.

sunilangadi2 commented on ceph/ceph_admin/dashboard.py:
```diff
@@ -0,0 +1,73 @@
+        "ceph mgr module enable dashboard",
+        "ceph dashboard create-self-signed-cert",
+        f"echo -n {pwd} > pass.txt",
+        f"ceph dashboard ac-user-create {user} -i pass.txt administrator",
```
@sunilkumarn417 Thanks for the review,
sure will work on the review comments.

sunilangadi2 commented:
@sunilkumarn417 Addressed review comments, Please review 
Success logs http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-FGXWY8/

sunilangadi2 commented on ceph/ceph_admin/dashboard.py:
```diff
@@ -0,0 +1,106 @@
+    sleep(100)
+    out, _ = cls.shell(args=["ceph", "mgr", "services"])
+    formatted = json.loads(out)
+    url = formatted["dashboard"]
```
Thanks done.

sunilangadi2 commented on ceph/ceph_admin/dashboard.py:
```diff
@@ -0,0 +1,106 @@
+
+
+def enable_dashboard(cls, configure_dashboard):
+    """
```
It's actually nice, Thanks for sharing the info
Done the changes as per the doc style.

sunilangadi2 commented:
@sunilkumarn417 Thanks for the review,
addressed all comments, please review
success logs:- http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1RNUOH/

sunilkumarn417 commented on tests/cephadm/test_dashboard.py:
```diff
@@ -0,0 +1,37 @@
+import logging
```
Add Module level doc string.

sunilkumarn417 commented on tests/cephadm/test_dashboard.py:
```diff
@@ -0,0 +1,37 @@
+        ceph_cluster (ceph.ceph.Ceph): Ceph cluster object
+        kw: test data
+
+    check ceph.ceph_admin.dashboard for test config
```
add `Returns` section in the doc-string.

psathyan commented:
@sunilangadi2 kindly let us know if you need any help in merging this PR... its been in open state for a while now.

sunilangadi2 commented:
@psathyan Initally i have addresed some review comments actually I need some more time to go through Google Style Python Docstrings https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html
Sunil kumar helped lot, Sure will finish it soon.
Thanks!

sunilangadi2 commented:
@sunilkumarn417 Addressed all doc corrections,
Please review.
